lib/charms/opensearch/v0/opensearch_relation_peer_cluster.py

```diff
@@ -519,6 +552,72 @@ def _fetch_local_cm_nodes(self, deployment_desc: DeploymentDescription) -> List[
             if node.is_cm_eligible() and node.app.id == deployment_desc.app.id
         ]
 
+    def _rel_data_redacted_dict(self, rel_data: PeerClusterRelData) -> dict[str, Any]:
+        """Convert the secret data to a dict for storage in a secret."""
+        # If we use the values of the secrets then the size of the secret
+        # would be too large for juju we store the secret ids instead and
+        # fetch the secrets when needed in the requirer side
+        redacted_dict = rel_data.to_dict()
```

Comment:
why is needed to serialize to a dict, why not work with the object and its attributes directly? I would rather we work with the former.

Reply:
We do not want to mutate the object directly. We can either use the dict or copy the object and update the content of the secret to their respective ids.

Reply:
@Mehdi-Bendriss We cannot directly use the `PeerClusterRelData` object directly because of the `admin_tls` field in the credentials which is `dict`. The secret IDs are `str`. I can either create a new model for the redacted peer cluster relation data or stick with the `dict`. WDYT?